GOVERNANCE_DOC.md

```diff
@@ -183,9 +177,83 @@ Additionaly, four lists will be linked to each proposal:
 - `InitVotingPowerList`: Snapshot of validators' voting power **when proposal enters voting period** (only saves validators whose voting power is >0).
 - `MinusesList`: List of minuses for each validator. Used to compute validators' voting power when they cast a vote.
 
+*Note: Actual data structure will probably not be list. These lists are here to illustrate what needs to be stored, not how*
+
 Two final parameters, `InitTotalVotingPower` and `InitProcedureNumber` associated with `proposalID` will be saved when proposal enters voting period.
 
 We also introduce `ProposalProcessingQueue` which lists all the `ProposalIDs` of proposals that reached `MinDeposit` from oldest to newest. Each round, the oldest element of `ProposalProcessingQueue` is checked during `BeginBlock` to see if `CurrentBlock == VotingStartBlock + InitProcedure.VotingPeriod`. If it is, then the application checks if validators in `InitVotingPowerList` have voted and, if not, applies `GovernancePenalty`. After that proposal is ejected from `ProposalProcessingQueue` and the new first element of the queue is evaluated. Note that if a proposal is urgent and accepted under the special condition, its `ProposalID` must be ejected from `ProposalProcessingQueue`.
+The beginning and the end of the `ProposalProcessingQueue` are identified using `ProposalProcessingQueueBeginning` and `ProposalProcessingQueueEnd`.
+
+And the pseudocode for the `ProposalProcessingQueue`:
+
+```
+  in BeginBlock do 
```

Comment:
I'm not sure we want pseudo code here, looks like it could just as easily be a series of bullet points (with sub bullet points) of procedure ordering... But if we do end up wanting pseudo code we should probably begin the code at `func checkProposal() ` and still used curly brackets around `funcs`, `if`, and `for` statements like in golang.

Reply:
Seeing that zarco did same formatting as me, I think we should keep it this way for now, so that it's uniform accross specs

Reply:
okay well remove the first few lines before `func checkProposal()` - that is implicit. Also I don't see how having the function name helps... just scanning through your pseudo code here, it's called within itself... which means that this is a recursive function... Is this what you wanted? - just want to make sure... we should probably be using the queue system instead of recursion.

Reply:
Talked with rigel, will use queue system+recursion. Fix to come